testing.rb

```diff
@@ -0,0 +1,389 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/BlockNesting,Metrics/MethodLength,Metrics/ModuleLength,Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity,Metrics/AbcSize
+
+# module Enumerable
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = to_a if self.class == Range
+    arr = self if self.class == Array
+    arr = self if self.class == Hash
+    n = 0
+    while n < arr.length
+      yield(arr[n])
+      n += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each) unless block_given?
+
+    arr = to_a if self.class == Range
+    arr = self if self.class == Array
+    arr = self if self.class == Hash
+    n = 0
+    while n < arr.length
+      yield(arr[n], n)
+      n += 1
+    end
+    self
+  end
```

Comment:
. #my_each_with_index when a block is given when self is a hash behaves like #each_with_index calls the given block once for each element in self